lua/telescope/sorters.lua

```diff
@@ -307,6 +307,49 @@ sorters.fuzzy_with_index_bias = function(opts)
   }
 end
 
+-- Sorter using the fzy algorithm
+sorters.get_fzy_sorter = function()
+  local fzy = require('telescope.algos.fzy')
+  local OFFSET = (fzy.MATCH_MAX_LENGTH + 1) * -fzy.SCORE_GAP_INNER
+
+  return sorters.Sorter:new{
+    scoring_function = function(_, prompt, line)
+      -- For some reason, the prompt is prefixed by a space. Strip
+      -- leading/trailing spaces from the prompt.
+      local prompt = string.match(prompt, '^%s*(.*%S)') or ""
+
+      -- Check for actual matches before running the scoring alogrithm.
+      if not fzy.has_match(prompt, line) then
+        return -1
+      end
+
+      local fzy_score = fzy.score(prompt, line)
+
+      -- The fzy score is -inf for empty queries and overlong strings.  Since
+      -- this function converts all scores into the range (0, 1), we can
+      -- convert these to 1 as a suitable "worst score" value.
+      if fzy_score == fzy.SCORE_MIN then
+        return 1
+      end
+
+      -- Poor non-empty matches can also have negative values. Offset the score
+      -- so that all values are positive, then invert to match the
+      -- telescope.Sorter "smaller is better" convention. Note that for exact
+      -- matches, fzy returns +inf, which when inverted becomes 0.
+      return 1 / (fzy_score + OFFSET)
+    end,
+
+    -- The fzy.positions function, which returns an array of string indices, is
+    -- compatible with telescope's conventions. It's moderately wasteful to
+    -- call call fzy.score(x,y) followed by fzy.positions(x,y): both call the
+    -- fzy.compute function, which does all the work. But, this doesn't affect
+    -- perceived performance.
```

Comment:
if you're interested in making them share the same computation, it is possible to do so. The first parameter to the function is `self` which can be used to store some intermediate state if you wish.

Otherwise we could do this some other time if we needed to.